lib/index.js

```diff
@@ -28,26 +43,53 @@ NamedURLResolverClass.prototype.escape = function(string) {
 
 NamedURLResolverClass.prototype.resolve = function(name, params) {
     if(name && (name in this.routesMap)) {
-        if(!params) params = {};
-
         var routePath = this.routesMap[name];
-        for(var paramName in params) {
-            if(params.hasOwnProperty(paramName)) {
-                let paramRegex = new RegExp('(/|^):' + paramName + '(/|$)');
-                let paramValue = this.escape('' + params[paramName]);
-                routePath = routePath.replace(paramRegex, '$1' + paramValue + '$2');
+
+        if(!params) {
+            return routePath;
+        }
+
+        for (var paramName in params) {
+            if (params.hasOwnProperty(paramName)) {
+                var paramValue = params[paramName];
+        
+                if (paramName === "splat") { // special param name in RR, used for "*" and "**" placeholders
+                    paramValue = toArray(paramValue); // when there are multiple globs, RR defines "splat" param as array.
+                    var i = 0;
+                    routePath = routePath.replace(reSplatParams, (match) => {
+                        var val = paramValue[i++];
+                        // don't escape slashes for double star, as "**" considered greedy by RR spec
+                        return val == null ? "" : this.escape('' + val, match === "*");
+                    });
+                } else {
+                    // Rougly resolve all named placeholders.
+                    // Cases: 
+                    // - "/path/:param"
+                    // - "/path/(:param)"
+                    // - "/path(/:param)"
+                    // - "/path(/:param/):another_param"
+                    // - "/path/:param(/:another_param)"
+                    // - "/path(/:param/:another_param)"
+                    var paramRegex = new RegExp('(\/|\\(|\\)|^):' + paramName + '(\/|\\)|\\(|$)');
+                    var replacement = '$1' + this.escape('' + paramValue) + '$2';
+                    routePath = routePath.replace(paramRegex, replacement);
+                }
             }
         }
-        return routePath;
+        return routePath
+            // Remove braces around resolved optional params (i.e. "/path/(value)")
```

Comment:
`(` and `)` may be a part of non-optional parameter value; I think the right way to approach this would be to match `(` and `)` but outside of capture group. Something like below maybe?
```js
var paramRegex = new RegExp('(?:[\\(]|(\/|^)):'+paramName+'(?:[\\)]|(\/|$)');
```